src/views/newConnection.tsx

```diff
@@ -1,42 +1,118 @@
 import Html from '@kitajs/html'
 import { singleton } from 'tsyringe'
 import { CompanyProfile } from '../models/companyHouseEntity'
-import { Page } from './common'
+import { FormButtonIcon, Page } from './common'
+
+export type CompanyProfileText =
+  | {
+      status: 'success'
+      company: CompanyProfile
+    }
+  | {
+      status: 'error'
+      errorMessage: string
+    }
+
+export type FormStage = 'form' | 'confirmation' | 'success'
 
 @singleton()
 export default class newConnectionTemplates {
   constructor() {}
 
-  public formPage = ({ targetBox }: { targetBox: JSX.Element }) => {
+  public formPage = (targetBox: CompanyProfileText, formStage: FormStage) => {
     return (
       <Page title="Veritable - New Connection" heading="New Connection" url="/connection/new">
-        <this.companyNumberInput targetBox={targetBox} />
+        <div class="connections header">
+          <span>Invite New Connection</span>
+        </div>
+        <this.companyFormInput targetBox={targetBox} formStage={formStage} />
       </Page>
     )
   }
 
-  public companyNumberInput = ({ targetBox }: { targetBox: JSX.Element }): JSX.Element => {
+  public companyFormInput = (params: {
+    targetBox: CompanyProfileText
+    formStage: FormStage
+    email?: string
+    companyNumber?: string
+  }): JSX.Element => {
+    const showForm = params.formStage === 'form' ? true : false
+    const showConfirmation = params.formStage === 'confirmation' ? true : false
+    const showSuccess = params.formStage === 'success' ? true : false
     return (
       <>
-        <form id="new-connection-form" class="input-group" hx-swap="outerHTML">
-          <input
-            id="company-number-input"
-            name="companyNumber"
-            placeholder="Company House Number"
-            required
-            hx-get="/connection/new/verify-company"
-            hx-trigger="change"
-            hx-target="#text-box-target"
-            hx-select="#text-box-target"
-            pattern="^(((AC|CE|CS|FC|FE|GE|GS|IC|LP|NC|NF|NI|NL|NO|NP|OC|OE|PC|R0|RC|SA|SC|SE|SF|SG|SI|SL|SO|SR|SZ|ZC|\d{2})\d{6})|((IP|SP|RS)[A-Z\d]{6})|(SL\d{5}[\dA]))$"
-            minlength={8}
-            maxlength={8}
-            oninput="this.reportValidity()"
-          ></input>
-          <input id="email" placeholder="Connection's Email Address" name="email"></input>
-          <button id="btn-cancel">Cancel</button>
-          <button id="btn-continue">Continue</button>
-          <div id="text-box-target">{targetBox}</div>
+        <form
+          id="new-connection-form"
+          class="new-connection"
```

Comment:
what does `new-connection` represent? It seems to be on a lot of elements with very different meaning. How bad would it be to remove all style classes?